uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -494,13 +509,21 @@ define([
       };
 
       /**
-       * Checks if given status is available to manager
+       * Checks if given status is available to manager. If manager applies leave
+       * on behalf of staff then cancelled is also removed from her list of available statuses.
        *
-       * @param {String} status
-       * @return {Boolean}
+       * @return {Function} that takes status as param and returns a boolean
        */
-      this.isStatusAvailableForManager = function (status) {
-        return !(status.name === 'admin_approved' || status.name === 'waiting_approval');
+      this.isStatusAvailableForManager = function () {
+        var self = this;
+
+        return function (status) {
```

Comment:
as we are passing paramter in filter , [more](http://stackoverflow.com/questions/27507114/how-to-pass-multiple-parameter-in-angular-filter-function-not-custom-filter)